tpmutil/encoding_test.go

```diff
@@ -247,9 +216,36 @@ func TestEncodingInvalidUnpack(t *testing.T) {
 
 }
 
+func TestSelfMarshaler(t *testing.T) {
+	var empty16 U16Bytes
+	var empty32 U32Bytes
+	subTests := []struct {
+		encoded []byte
+		decoded interface{}
+	}{
+		{[]byte{0, 0}, &empty16},
+		{[]byte{0, 1, 137}, &empty16},
+		{[]byte{0, 0, 0, 0}, &empty32},
+		{[]byte{0, 0, 0, 1, 137}, &empty32},
+	}
+	for _, st := range subTests {
```

Comment:
either use subtests (`t.Run`) or print input for each call that fails below.
otherwise it's really hard to figure out which input failed

Reply:
Added Logf